pkg/service/model_build_load_balancer.go

```diff
@@ -217,14 +241,27 @@ func (t *defaultModelBuildTask) getMatchingIPforSubnet(_ context.Context, subnet
 	return "", errors.Errorf("no matching ip for subnet %s", *subnet.SubnetId)
 }
 
-func (t *defaultModelBuildTask) resolveLoadBalancerSubnets(ctx context.Context, scheme elbv2model.LoadBalancerScheme) ([]*ec2.Subnet, error) {
+func (t *defaultModelBuildTask) buildLoadBalancerSubnets(ctx context.Context, scheme elbv2model.LoadBalancerScheme) ([]*ec2.Subnet, error) {
 	var rawSubnetNameOrIDs []string
 	if exists := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixSubnets, &rawSubnetNameOrIDs, t.service.Annotations); exists {
 		return t.subnetsResolver.ResolveViaNameOrIDSlice(ctx, rawSubnetNameOrIDs,
 			networking.WithSubnetsResolveLBType(elbv2model.LoadBalancerTypeNetwork),
 			networking.WithSubnetsResolveLBScheme(scheme),
 		)
 	}
+
+	existingLB, err := t.fetchExistingLoadBalancer(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if existingLB != nil {
+		subnetIDs := make([]string, 0, len(existingLB.LoadBalancer.AvailabilityZones))
+		return t.subnetsResolver.ResolveViaNameOrIDSlice(ctx, subnetIDs,
```

Comment:
You need to iterate through the subnets in the existingLB.LoadBalancer.AvailabilityZones and push the subnetIDs to the list.

Reply:
Changed, thanks!